backends/bmv2/common/controlFlowGraph.h

```diff
@@ -72,10 +72,12 @@ class CFG final : public IHasDbPrint {
         const cstring name;
         EdgeSet successors;
 
-        void dbprint(std::ostream &out) const;
+        void dbprint(std::ostream &out) const override;
         void addPredecessors(const EdgeSet *set);
         void computeSuccessors();
         cstring toString() const { return name; }
+
+        DECLARE_TYPEINFO(Node);
```

Comment:
Missing `ICastable` base? While I don't see many reasons to cast to `ICastable`, I think the typeid declarations should match the hierarchy (restricted to what is `RTTI::Base`-derived). Otherwise we risk very weird bugs.

Reply:
Actually not, exactly the opposite. Here is how we are declaring that `Node` is the base of our class hierarchy. We do not have bare `ICastable` pointers, we are not supposed to cast to `ICastable`. Even if we'd try to do this, we'll end with compile errors (as `ICastable` does not have RTTI).

So, the typeid lookup for `is<T>` will end with `Node` as well as all casts

Reply:
I see, so basically this separates the various RTTI-enabled hierarchies from each other, right? We don't want to have pointers to `RTTI::Base` floating around. Could you please add a comment in into `rtti.h` about this.

Reply:
Yes, exactly.